src/popup/components/TransactionOverview.vue

```diff
@@ -8,103 +8,131 @@
+              && (props.tx.accountId || props.tx.callerId),
+          );
           const contract = {
-            address: this.tx.contractId,
-            url: this.getExplorerPath(this.tx.contractId),
-            label: this.$t(`transaction.overview.${this.isDexRecipient ? 'superheroDex' : 'contract'}`),
+            address: props.tx.contractId,
+            url: getExplorerPath.value(props.tx.contractId),
+            label: root.$t(`transaction.overview.${isDexRecipient.value ? 'superheroDex' : 'contract'}`),
           };
           return {
-            sender: direction === 'sent' ? this.account : contract,
-            recipient: direction === 'received' ? this.account : contract,
-            title: this.$t('transaction.type.contractCallTx'),
-            function: this.tx.function,
+            sender: direction === TRANSACTION_DIRECTION_SENT
+              ? ownershipAccount.value
+              : contract,
+            recipient: direction === TRANSACTION_DIRECTION_RECEIVED
+              ? ownershipAccount.value
+              : contract,
+            title: root.$t('transaction.type.contractCallTx'),
+            function: props.tx.function,
           };
         }
         case SCHEMA.TX_TYPE.contractCreate:
           return {
-            sender: this.account,
+            sender: ownershipAccount.value,
             recipient: {
               contractCreate: true,
-              label: this.$t('transaction.overview.contractCreate'),
+              label: root.$t('transaction.overview.contractCreate'),
             },
-            title: this.$t('transaction.type.contractCreateTx'),
+            title: root.$t('transaction.type.contractCreateTx'),
           };
         case SCHEMA.TX_TYPE.namePreClaim:
         case SCHEMA.TX_TYPE.nameClaim:
         case SCHEMA.TX_TYPE.nameBid:
         case SCHEMA.TX_TYPE.nameUpdate:
           return {
-            sender: this.account,
+            sender: ownershipAccount,
             recipient: {
               aens: true,
-              label: this.$t('transaction.overview.aens'),
+              label: root.$t('transaction.overview.aens'),
             },
-            title: this.$t('transaction.type')[this.txType],
+            title: transactionTypes[txType.value],
           };
         default:
           throw new Error('Unsupported transaction type');
       }
-    },
-    txType() {
-      return this.getTxType({ tx: this.tx });
-    },
-  },
-  async mounted() {
-    await watchUntilTruthy(() => this.$store.state.middleware);
-    if (this.tx.recipientId?.startsWith('nm_')) {
-      this.name = (await this.$store.state.middleware.getNameById(this.tx.recipientId)).name;
-    }
+    });
+
+    onMounted(async () => {
+      await watchUntilTruthy(() => middleware.value);
+      if (props.tx.recipientId?.startsWith('nm_')) {
+        name.value = (await middleware.value.getNameById(props.tx.recipientId)).name;
```

Comment:
recipientId what if recipientId not exist?

Reply:
then it won't start with 'nm_' :)